src/book/03-guides/04-general/04-code-conventions.mdx

```diff
@@ -1,12 +1,48 @@
 ---
 section: Guides
 chapter: General
-title: Code Documentation
-description: How to use documentation tags
-slug: /guides/general/documentation
+title: Code Conventions
+description: Conventions used in NUbots code.
+slug: /guides/general/code-conventions
 ---
 
-## Conventions
+## Naming Conventions
+
+### Case
+
+NUbots code case conventions are in the below table. It does not include JavaScript, which is in the following table.
+
+| Scenario      | Case Type            | Example                |
+| ------------- | -------------------- | ---------------------- |
+| Class name    | PascalCase           | `MyClass`              |
+| Function name | snake_case           | `my_function()`        |
+| Variable name | snake_case           | `my_variable`          |
+| Constants     | SCREAMING_SNAKE_CASE | `MY_CONSTANT_VARIABLE` |
+| `yaml` fields | snake_case           | `my_field: 0.0`        |
+
+JavaScript conventions are in the below table.
+
+| Scenario      | Case Type            | Example                |
+| ------------- | -------------------- | ---------------------- |
+| Class name    | PascalCase           | `MyClass`              |
+| Function name | camelCase            | `myFunction()`         |
+| Variable name | camelCase            | `myVariable`           |
+| Constants     | SCREAMING_SNAKE_CASE | `MY_CONSTANT_VARIABLE` |
+
+### Mathematics
+
+Conventions for variables representing vectors and homogeneous transformations are explained in [the Linear Algebra section of the Mathematics page](/system/foundations/mathematics#linear-algebra).
+
+A summary is given in the table below.
+
+| Variable name | Description                                                                  |
+| ------------- | ---------------------------------------------------------------------------- |
+| `Hab`         | Denotes a 3D affine transformation matrix going from space $b$ to space $a$. |
+| `Rab`         | Denotes a 3D rotation matrix from space $b$ to space $a$.                    |
+| `rABb`        | Denotes a vector from point $B$ to point $A$ in space $b$.                   |
+| `vAb`         | The velocity of $A$ in space $b$.                                            |
```

Comment:
I mean, if we're going to include velocity as a thing here (which I have never noticed), should we also include things like _omegaAAb_ for rotational velocity? 

We should also specify that `Rab` can be a quaternion (e.g., see motionmodel)

Reply:
What are the two A's in _omegaAAb_? 😵 @TrentHouliston